src/spaces/openstreetmap.jl

```diff
@@ -0,0 +1,402 @@
+export OpenStreetMapSpace, OSMPos, OSMSpace
+export osm_random_direction, osm_plan_route, osm_map_coordinates, osm_road_length
+
+struct OpenStreetMapSpace <: DiscreteSpace
+    m::OpenStreetMapX.MapData
+    edges::Vector{CartesianIndex{2}} #So far, only helpful for osm_random_direction
+    s::Vector{Vector{Int}}
+end
+
+"""
+    OpenStreetMapSpace(path::AbstractString; kwargs...)
+Create a space residing on the Open Street Map (OSM) file provided via `path`.
+
+The abbreviation `OSMSpace` may be used interchangably.
+
+Much of the functionality of this space is provided by interfacing with
+[OpenStreetMapX.jl](https://github.com/pszufe/OpenStreetMapX.jl), for example the two
+keyword arguments `use_cache = false` and `trim_to_connected_graph = true` are
+passed into the [`OpenStreetMapX.get_map_data`](@ref) function.
+For details on how to obtain an OSM file for your use case, consult the OpenStreetMapX.jl
+Readme.
+"""
```

Comment:
The docstring of `OpenStreetMapSpace` should explain how the business of of planning a route works. Furthermore, I believe that the expected `Agent` fields and their types should be documented here, in this docstring, instead of the `AbstractAgent` docstring.

Reply:
Oh, also we should discuss here that this space really represents the map as a _continuous_ entity, in the expence of performance. if one only cares about interesections they can just load to graph and make a `GraphSpace`.

Reply:
Good with all of those things. Was more focused on implementation. Will keep this as a TODO once everything else settles down here.